discovery/data_sharing/v0.1/data_sharing.md

```diff
@@ -0,0 +1,297 @@
+to save bandwidth only URIs of content are actually exchanged with the
+FASP. The FASP is then responsible for retrieving the content and
+working with it. The exact details of this differ between capabilties
+and are described in the respective sections.
+
+A fediverse server MUST make sure it only shares content that it is
+actually allowed to share. It MUST NOT share any content that is not
+public. In the case of account and post data it MUST make sure to only
+share content where the creator has opted in to discovery.
+
+The FASP in turn MUST also ensure that creators have opted in to
+discovery before storing and indexing content.
+
+Both parties MUST use the mechanism described in
+[FEP-5feb](https://codeberg.org/fediverse/fep/src/branch/main/fep/5feb/fep-5feb.md)
+to determine if a creator has opted in to discovery.
+
+For account data, both parties MUST ensure the `discoverable` flag is
+set to `true`.
+
+## Subscribing To, Requesting And Receiving Content
+
+### Managing Subscriptions (FASP => Fediverse Server)
+
+In order to subscribe to new content, FASPs can subscribe to events
+by making a `POST` call to the `/data_sharing/v0/event_subscriptions`
+endpoint on the fediverse server.
+
+Example call:
+
+```http
+POST /data_sharing/v0/event_subscriptions
+```
+
+The request body MUST contain a JSON object defining what events to
+subscribe to. This object MUST contain the keys `objectType` and
+`eventType`. It MAY contain the keys `maxBatchSize` and/or
+`threshold`. The keys are defined as follows:
+
+* `category`: One of `content`, `account`. This is the category of
+  objects the FASP is interested in.
+* `subscriptionType`: One of `lifecycle`, `trends`. `lifecycle`
+  means the FASP would like to be notified when new objects of the given
+  type are created, when objects are updated and when objects are
+  deleted. `trends` means that the FASP would like to be notified of
+  objects that had recent interactions. `trends` is only applicable if
+  the `category` given is `content`.
+* `maxBatchSize`: The maximum number of events the FASP would like to
+  receive in a single request. This is optional.
+* `threshold`: If `eventType` is `trending` then this object can be used to
+  further specify when the event should be reported. Valid keys for this
+  object are:
+  * `timeframe`: Number of minutes in which interactions should fire an
+    event, defaults to `15`
+  * `shares`: Number of shares in the given timeframe, defaults to `3`
+  * `likes`: Number of likes in the given timeframe, defaults to `3`
+  * `replies`: Number of replies in the given timeframe, defaults to `3`
+
+Example object:
+
+```json
+{
+  "category": "content",
+  "subscriptionType": "trends",
+  "maxBatchSize": 10,
+  "threshold": {
+    "timeframe": 15,
+    "shares": 3
+  }
+}
+```
+
+The fediverse server MUST validate the request. If it is invalid it MUST
+return an HTTP status code `422` (Unprocessable Content).
```

Comment:
Should it return different errors for different things? e.g., 400 if the payload isn't just, 422 if the schema isn't correct, and 401 if the http signatures don't validate?

Reply:
The latter is already part of the "general" spec. Do we need the distinction between 400 and 422? I have no idea :confused:

Reply:
There's possibly also 429, or 409 (conflict) for when a FASP already has a subscription that matches this subscription.